lib/active_storage/service/mirror_service.rb

```diff
@@ -26,9 +26,8 @@ def primary_service
     end
 
     def perform_across_services(method, *args)
-      # FIXME: Convert to be threaded
       services.collect do |service|
-        service.public_send method, *args
-      end
+        Thread.new { service.public_send method, *args }
+      end.map(&:join)
```

Comment:
Can use concurrent-ruby's promises (or a lower level thread pool executor). Check out its usage in Action Cable.